### PR TITLE
nrf52: Add suspend/resume detection to usbdev

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds USB suspend/resume state detection to the nrf52840 usbdev peripheral. This is essential for detecting when the host side suspends bus activity and thus when not to expect any periodic transfer to be serviced

### Testing procedure

1. Enable debug in `sys/usb/usbus/usbus.c` to print USB suspend/resume info.
2. flash on one of the nrf52840 based boards
3. Plug the device usb port in a Linux based host (needed for next steps)
4. Find out which usb port is used, `dmesg` usually shows this info, should be something along the lines of `3-9.4`, depends on the usb bus, hub and port it is plugged into.
5. use:
```
echo "0" > "/sys/bus/usb/devices/$PORT/power/autosuspend"
echo "auto" > "/sys/bus/usb/devices/$PORT/power/level"
```
to force the port in suspend mode.

6. Verify that the device prints `usbus: USB suspend detected`
7. use:
```
echo "on" > "/sys/bus/usb/devices/$PORT/power/level"
```
To resume the device

8. Verify that the device prints `usbus: USB resume detected`
9. Verify that the messages are also printed when unplugging and plugging in the device.

### Issues/PRs references

None

### CODE REVIEW

File: cpu/nrf52/periph/usbdev.c [hunk 1]
```diff
@@ -539,6 +541,19 @@ static void _esr(usbdev_t *dev)
...
+        uint32_t events = usbdev->device->EVENTCAUSE;
+        if (events & USBD_EVENTCAUSE_SUSPEND_Msk) {
+            usbdev->usbdev.cb(&usbdev->usbdev, USBDEV_EVENT_SUSPEND);
+        }
+        if (events & USBD_EVENTCAUSE_RESUME_Msk) {
+            usbdev->usbdev.cb(&usbdev->usbdev, USBDEV_EVENT_RESUME);
+        }
+        usbdev->device->EVENTS_USBEVENT = 0;
+        /* Clear eventcause register */
+        usbdev->device->EVENTCAUSE = 0x0f01;
```
Comment (aabadie) [inline cpu/nrf52/periph/usbdev.c]:
```suggestion
        usbdev->device->EVENTCAUSE = (USBD_EVENTCAUSE_READY_Msk |
                                      USBD_EVENTCAUSE_USBWUALLOWED_Msk |
                                      USBD_EVENTCAUSE_RESUME_Msk |
                                      USBD_EVENTCAUSE_SUSPEND_Msk |
                                      USBD_EVENTCAUSE_ISOOUTCRC_Msk);
```

File: cpu/nrf52/periph/usbdev.c [hunk 2]
```diff
@@ -275,7 +275,9 @@ static void _init(usbdev_t *dev)
     usbdev->sstate = NRFUSB_SETUP_READY;
 
     /* Enable a set of interrupts */
-    usbdev->device->INTEN = USBD_INTEN_USBRESET_Msk | USBD_INTEN_EPDATA_Msk;
+    usbdev->device->INTEN = USBD_INTEN_USBRESET_Msk |
```
Comment (aabadie) [inline cpu/nrf52/periph/usbdev.c]: it's better if you put parenthesis around mask values:
```suggestion
    usbdev->device->INTEN = (USBD_INTEN_USBRESET_Msk |
                             USBD_INTEN_EPDATA_Msk |
                             USBD_INTEN_USBEVENT_Msk);
```

File: cpu/nrf52/periph/usbdev.c (hunk 1 above)
Comment (bergzand) [reply cpu/nrf52/periph/usbdev.c]: Fixed

File: cpu/nrf52/periph/usbdev.c (hunk 2 above)
Comment (bergzand) [reply cpu/nrf52/periph/usbdev.c]: Also fixed

**Comment by aabadie** [comment]
and go!

**Comment by bergzand** [comment]
Thank you for reviewing!

**Pull request merged**